### PR TITLE
fix(devices): remove plural strings from connected date

### LINKED ISSUE #3510: Update connected date to exclude plurals
I sold @ryanfeeley on this idea:

to avoid the plurals we are going to modify our strings, but @ryanfeeley also wanted to add "weeks" into the mix:

## Before

```
 +  _.extend($.timeago.settings.strings, {
 +    day: t('a day'),
 +    days: t('%d days'),
 +    hour: t('about an hour'),
 +    hours: t('%d hours'),
 +    minute: t('about a minute'),
 +    minutes: t('%d minutes'),
 +    month: t('about a month'),
 +    months: t('%d months'),
 +    seconds: t('a few seconds'),
 +    suffixAgo: t('ago'),
 +    suffixFromNow: t('from now'),
 +    year: t('about a year'),
 +    years: t('%d years')
 +  });
```

## After

```
 +  _.extend($.timeago.settings.strings, {
 +    day: t('a day ago'),
 +    days: t('a few days ago'),
+     week: t('a week ago'),
 +    weeks: t(' a few weeks ago'),
 +    hour: t('about an hour ago'),
 +    hours: t(' a few hours ago'),
 +    minute: t('about a minute ago'),
 +    minutes: t('a few minutes ago'),
 +    month: t('about a month ago'),
 +    months: t(' a few months ago'),
 +    seconds: t('a few seconds ago'),
 +    suffixAgo: t(''),
 +    suffixFromNow: t('from now'), // remove this??
 +    year: t('about a year ago'),
 +    years: t('a few years ago')
 +  });
```
Added weeks

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #3510
Fixes https://github.com/mozilla/fxa-content-server-l10n/issues/104

### CODE REVIEW

**Comment by vladikoff** [comment]
@shane-tomlinson r?

File: app/scripts/views/settings/devices.js [hunk 1]
```diff
@@ -24,19 +24,20 @@ define(function (require, exports, module) {
...
+    day: t('a day ago'),
+    days: t('a few days ago'),
+    hour: t('about an hour ago'),
+    hours: t('a few hours ago'),
+    minute: t('about a minute ago'),
+    minutes: t('a few minutes ago'),
+    month: t('about a month ago'),
+    months: t('a few months ago'),
+    seconds: t('a few seconds ago'),
+    suffixAgo: '',
```
Comment (shane-tomlinson) [inline app/scripts/views/settings/devices.js]: Is this needed?

Comment (vladikoff) [inline app/scripts/views/settings/devices.js]: yeah, otherwise the default behaviour will add "a week ago ago"

Comment (shane-tomlinson) [inline app/scripts/views/settings/devices.js]: he, good call then.

File: app/scripts/views/settings/devices.js [hunk 2]
```diff
@@ -24,19 +24,20 @@ define(function (require, exports, module) {
...
+    week: t('a week ago'),
+    weeks: t('a few weeks ago'),
+    year: t('about a year ago'),
+    years: t('a few years ago')
```
Comment (shane-tomlinson) [inline app/scripts/views/settings/devices.js]: I'm looking at your branch, this string is used for > 1.5 years. Maybe this should be `more than a year ago`

File: app/scripts/views/settings/devices.js [hunk 3]
```diff
@@ -24,19 +24,20 @@ define(function (require, exports, module) {
...
-    year: t('about a year'),
-    years: t('%d years')
+    day: t('a day ago'),
+    days: t('a few days ago'),
+    hour: t('about an hour ago'),
+    hours: t('a few hours ago'),
+    minute: t('about a minute ago'),
+    minutes: t('a few minutes ago'),
+    month: t('about a month ago'),
+    months: t('a few months ago'),
```
Comment (shane-tomlinson) [inline app/scripts/views/settings/devices.js]: This string is used if days > 45 but less than 365. Perhaps this should be `more than a month ago`

File: app/scripts/views/settings/devices.js [hunk 4]
```diff
@@ -24,19 +24,20 @@ define(function (require, exports, module) {
...
-    days: t('%d days'),
-    hour: t('about an hour'),
-    hours: t('%d hours'),
-    minute: t('about a minute'),
-    minutes: t('%d minutes'),
-    month: t('about a month'),
-    months: t('%d months'),
-    seconds: t('a few seconds'),
-    suffixAgo: t('ago'),
-    suffixFromNow: t('from now'),
```
Comment (shane-tomlinson) [inline app/scripts/views/settings/devices.js]: You added `suffixAgo`, but not `suffixFromNow` in the new strings. Intentional?

**Comment by shane-tomlinson** [comment]
@ryanfeeley - any suggestions for the text?

**Comment by shane-tomlinson** [comment]
@vladikoff - other than the questions about text and `suffixFromNow`, looks good.

File: app/scripts/views/settings/devices.js (hunk 2 above)
Comment (vladikoff) [inline app/scripts/views/settings/devices.js]: :+1:

File: app/scripts/views/settings/devices.js (hunk 4 above)
Comment (vladikoff) [inline app/scripts/views/settings/devices.js]: FromNow is only for future stuff, we don't use that

Comment (ryanfeeley) [inline app/scripts/views/settings/devices.js]: @vladikoff No weeks as discussed?

Comment (vladikoff) [inline app/scripts/views/settings/devices.js]: @ryanfeeley weeks are fine

**Comment by rfk** [comment]
I like this move.  Do we need to get some l10n folks into this PR for f? or are we comfortable that we're on the right track thanks to out-of-band conversations?

**Comment by ryanfeeley** [comment]
@rfk IMHO it always makes sense to sanity check i18n. I always get burned when I make assumptions about other languages. Remember when I learned that Czech can't express decades consisely like 1980s?

**Comment by rfk** [comment]
TIL that Czech can't express decades concisely...

File: app/scripts/views/settings/devices.js (hunk 3 above)
Comment (ryanfeeley) [inline app/scripts/views/settings/devices.js]: Thinking about how people state their baby age in weeks and months: is less more in this case?

1. <120 seconds = seconds ago
1. <2 minutes = minutes ago
1. <120 minutes = hours ago
1. <48 hours = days ago
1. <14 days = weeks ago
1. <8 weeks = months ago
1. <24 months = years ago

**Comment by splewako** [comment]
How exactly do you plan to use changed ("days ago", "hours ago", "minutes ago", "months ago", "seconds ago", "weeks ago" and "years ago") strings?

**Comment by vladikoff** [comment]
> How exactly do you plan to use changed ("days ago", "hours ago", "minutes ago", "months ago", "seconds ago", "weeks ago" and "years ago") strings?

It will be under the device name in the devices view. Screenshot: https://github.com/mozilla/fxa/tree/master/features/FxA-16-devices-view#minimal-view-no-rename

**Comment by splewako** [comment]
>>    How exactly do you plan to use changed ("days ago", "hours ago", "minutes ago", "months ago", "seconds ago", "weeks ago" and "years ago") strings?
> 
> It will be under the device name in the devices view. Screenshot: https://github.com/mozilla/fxa/tree/master/features/FxA-16-devices-view#minimal-view-no-rename

Thank you. We should be able to come up with something quite nice (@piotrdrag) assuming the screenshot with numbers is old and the strings will not get used anywhere else.

**Comment by TheoChevalier** [comment]
@vladikoff not sure if I read the code right, but it seems we are now using the same string for both singular and plural and that variables have been removed from the strings?
Even in English, as I read the code, it seems we can't display "1 day ago", "1 hour ago" anymore, but "1 days ago", "1 hours ago" instead?

Please let me know if I'm missing something obvious, but as a localizer I currently have no clue about how I can translate those strings in a similar way to this: https://github.com/rmm5t/jquery-timeago/blob/master/locales/jquery.timeago.fr.js
Side note, if localizers are allowed to add back a %d variable in the translated strings to move the number part, it should be mentioned in a comment as they won't dig into the code. As you can see in the French example above, some locales need to move the number into the string to be able to translate it properly.

**Comment by TheoChevalier** [comment]
Nevermind, I didn't understand that you completely removed the number part.

**Pull request merged**